### PR TITLE
Report last attempt as error if it would be handled

### PR DESCRIPTION
**Pull request by peter-csala**
# Pull Request

- #2162

## Details on the issue fix or feature implementation
- If the retry exhausted all of its attempts then report the `ExecutionAttempt` as `Error` if it would handle it
- If the retry exhausted all of its attempts then report the `ExecutionAttempt` as `Warning` if it would not handle it

## Missing
- Update documentation 

## Out of scope
- This PR does not include changes for hedging strategy

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
> Update documentation

Are you planning on adding to this PR, or in another one?

**Comment by peter-csala** [comment]
> > Update documentation
> 
> Are you planning on adding to this PR, or in another one?

Yes, I will do that tomorrow morning.

**Comment by peter-csala** [comment]
@martincostello Could you please re-run the `build / windows-latest` step?

> [xUnit.net 00:00:02.90]     Polly.Specs.Utilities.SystemClockSpecs.SleepAsync_ShouldThrow_WhenCancellationRequested [FAIL]

This test seems a bit flaky :(

**Comment by martincostello** [comment]
You don't have to ask any time that happens, I'll just do it anyway, as otherwise it's unmergeable.

File: docs/strategies/retry.md [hunk 1]
```diff
@@ -167,6 +167,8 @@ Execution attempt. Source: 'MyPipeline/MyPipelineInstance/MyRetryStrategy', Oper
 > On the other hand the `Execution attempt` event will be **always** reported regardless whether the strategy has to perform any retries.
 >
 > Also remember that `Attempt: '0'` relates to the original execution attempt.
+>
+> Only the very last error event will have `Error` severity if it was unsuccessful.
```
Comment (martincostello) [inline docs/strategies/retry.md:171]:
```suggestion
> Only the last error event will have `Error` severity if it was unsuccessful.
```

**Review by martincostello — APPROVED** [review]

**Comment by peter-csala** [comment]
@martincostello , @martintmk What do you guys think, would it make sense to implement the same for hedging?

**Pull request merged**